prov/lpp/include/lpp.h

```diff
@@ -0,0 +1,231 @@
+ *     conditions are met:
+ *
+ *      - Redistributions of source code must retain the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer.
+ *
+ *      - Redistributions in binary form must reproduce the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer in the documentation and/or other materials
+ *        provided with the distribution.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#ifndef _LPP_H_
+#define _LPP_H_
+
+#include <stdio.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <stdatomic.h>
+#include <unistd.h>
+#include <string.h>
+#include <search.h>
+#include <pthread.h>
+#include <sys/ioctl.h>
+#include <stdbool.h>
+#include <netdb.h>
+#include <netinet/in.h>
+#include <sys/uio.h>
+#include <config.h>
+
+#include <ofi.h>
+#include <ofi_atom.h>
+#include <ofi_atomic.h>
+#include <ofi_mr.h>
+#include <ofi_enosys.h>
+#include <ofi_indexer.h>
+#include <ofi_iov.h>
+#include <ofi_rbuf.h>
+#include <ofi_list.h>
+#include <ofi_lock.h>
+#include <ofi_file.h>
+#include <ofi_osd.h>
+#include <ofi_util.h>
+#include <ofi_hmem.h>
+
+#include "linux/klpp/klpp_shared.h"
+#include "linux/klpp/klpp_ioctl.h"
+#include "lpp_jhash.h"
+
+#ifndef compiler_barrier
+#define compiler_barrier() asm volatile ("" ::: "memory")
+#endif
+
+#ifndef ARRAY_SIZE
+#define ARRAY_SIZE(x) (sizeof(x) / sizeof((x)[0]))
+#endif
+
+//
+// Bounds constants.
+//
+#define LPP_MR_CNT (256)
+
+#define LPP_CQ_MAX_COUNT (1024)
+#define LPP_CQ_MAX_SIZE (65536)
+#define LPP_CQ_DEFAULT_SIZE (65536)
+
+#define LPP_EP_MAX_COUNT (256)
+#define LPP_TX_CTX_MAX_COUNT (4096)
+#define LPP_RX_CTX_MAX_COUNT (4096)
+#define LPP_CNTR_MAX_COUNT (16 * 1024)
+#define LPP_MIN_MULTI_RECV (64)
+#define LPP_MAX_INJECT (64)
+
+#if HAVE_CUDA || HAVE_ROCR
+#define LPP_CAPS_HMEM FI_HMEM
+#else
+#define LPP_CAPS_HMEM 0
+#endif
+
+//
+// Per fi_getinfo(3):
+//
+// Capabilities may be grouped into two general categories: primary and
+// secondary. Primary capabilities must explicitly be requested by an
+// application, and a provider must enable support for only those primary
+// capabilities which were selected. Secondary capabilities may optionally be
+// requested by an application. If requested, a provider must support the
+// capability or fail the fi_getinfo request (FI_ENODATA). A provider may
+// optionally report non-selected secondary capabilities if doing so would not
+// compromise performance or security.
+//
```

Comment:
Please use `/*` `*/` for multiline comments. Same for many other places.

Reply:
done